File: VB/ReportingEUDSaveReportWithoutConnectionParams/CustomSqlDataConnection.vb
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Native
Imports DevExpress.DataAccess.Sql

Namespace WinEUDSaveReportWithoutConnectionParams
    Friend Class CustomSqlDataConnection
        Inherits SqlDataConnection
        Implements INamedItem

        Public Sub New(ByVal name As String, ByVal connectionParameters As DataConnectionParametersBase)
            MyBase.New(name, connectionParameters)
        End Sub

        Private Property INamedItem_Name() As String Implements INamedItem.Name
            Get
                Return Name & " (Custom)"
            End Get
            Set(ByVal value As String)
                Name = value
            End Set
        End Property
    End Class
End Namespace

File: CS/ReportingEUDSaveReportWithoutConnectionParams/CustomSqlDataConnection.cs
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Native;
using DevExpress.DataAccess.Sql;

namespace WinEUDSaveReportWithoutConnectionParams {
    class CustomSqlDataConnection : SqlDataConnection, INamedItem {
        public CustomSqlDataConnection(string name, DataConnectionParametersBase connectionParameters)
            : base(name, connectionParameters) {
        }

        string INamedItem.Name {
            get {
                return Name + " (Custom)";
            }
            set {
                Name = value;
            }
        }
    }
}

